types/styled-components/index.d.ts

```diff
@@ -436,4 +436,24 @@ export class StyleSheetManager extends React.Component<
     StyleSheetManagerProps
 > {}
 
+export type CSSIntrinsicAttributeType =
+    | string
+    | CSSObject
+    | FlattenSimpleInterpolation
+    // Sad, but because this is global, there is no way to override it with the ThemedStyledComponentsModule
+    // Only augmenting DefaultTheme will work for inline css prop
+    | FlattenInterpolation<ThemeProps<AnyIfEmpty<DefaultTheme>>>;
+
 export default styled;
+
+declare module "react" {
+    interface Attributes {
+        // NOTE: unlike the plain javascript version, it is not possible to get access
+        // to the element's own attributes inside function interpolations.
+        // Only theme will be accessible, and only with the DefaultTheme due to the global
+        // nature of this declaration.
+        // If you are writing this inline you already have access to all the attributes anyway,
+        // no need for the extra indirection.
+        css?: import(".").CSSIntrinsicAttributeType; // tslint:disable-line whitespace
```

Comment:
Instead of having this odd import here, couldn't you put the `export type CSSIntrinsicAttributeType` inside the `"react"` module, reference it here directly, and import it from `"react"` in other places?

Reply:
If I moved it there, I'd still have to use the import() to be able to access FlattenInterpolation, CSSObject et al, while creating an actual type name in another module that only exists if this one is imported.

Reply:
I was actually slightly confused by not being able to just refer to the name as if it was in scope, having to add an import. There's probably something odd about how react itself is declared that prevents it, maybe the namespace being the export.

Reply:
Interestingly, I was able to just refer to `CSSIntrinsicAttributeType` in scope this time.

I wonder if it broke when I tried to put it inside `namespace React` (which broke other things too).